packages/preferences/src/browser/preferences-contribution.ts

```diff
@@ -49,33 +45,30 @@ export class PreferencesContribution extends AbstractViewContribution<Preference
         });
     }
 
-    registerCommands(commands: CommandRegistry): void {
-        commands.registerCommand(PREFERENCES_COMMAND, {
+    async registerCommands(commands: CommandRegistry): Promise<void> {
+        commands.registerCommand(CommonCommands.OPEN_PREFERENCES, {
             isEnabled: () => true,
-            execute: () => this.openPreferences()
+            execute: (preferenceScope = PreferenceScope.User) => this.openPreferences(preferenceScope)
         });
     }
 
     registerMenus(menus: MenuModelRegistry): void {
         menus.registerMenuAction(CommonMenus.FILE_SETTINGS_SUBMENU_OPEN, {
-            commandId: PREFERENCES_COMMAND.id,
+            commandId: CommonCommands.OPEN_PREFERENCES.id,
             order: 'a10'
         });
     }
 
-    protected async openPreferences(): Promise<void> {
-        const userUri = this.userPreferenceProvider.getUri();
-        const content = await this.userStorageService.readContents(userUri);
-        if (content === '') {
-            await this.userStorageService.saveContents(userUri, '');
-        }
-
+    protected async openPreferences(preferenceScope: PreferenceScope): Promise<void> {
         const wsUri = await this.workspacePreferenceProvider.getUri();
         if (wsUri && !await this.filesystem.exists(wsUri.toString())) {
             await this.filesystem.createFile(wsUri.toString());
         }
 
+        const widget = await this.widget;
+        widget.preferenceScope = preferenceScope;
         super.openView({ activate: true });
+        widget.activatePreferenceEditor(preferenceScope);
```

Comment:
I think it should be an internal API, and be called `onActivate` callback by the widget itself, i.e. from the client perspective calling `widget.preferenceScope = preferenceScope` should be enough, the client should not care to calls something else. I'm fine to change it later though.

Reply:
the only reason I was calling it also at this point was so that if the preferences widget was already open (`user preferences`), and the user attempted to add a new cpp config, the editor would toggle to the `workspace preferences` where they can see the new configuration.